connectors/src/connectors/microsoft/temporal/activities.ts

```diff
@@ -540,6 +543,39 @@ export async function syncDeltaForRootNode({
   );
 }
 
+export async function microsoftGarbageCollectorActivity({
+  connectorId,
+  nodeIdsToDelete,
+}: {
+  connectorId: number;
+  nodeIdsToDelete: string[];
+}) {
+  const connector = await ConnectorResource.fetchById(connectorId);
+  if (!connector) {
+    throw new Error(`Connector ${connectorId} not found`);
+  }
+
+  const dataSourceConfig = dataSourceConfigFromConnector(connector);
+
+  for (const nodeId of nodeIdsToDelete) {
+    const node = await MicrosoftNodeResource.fetchByInternalId(
+      connectorId,
+      nodeId
+    );
+    if (node) {
+      try {
+        await deleteFromDataSource(dataSourceConfig, node.internalId);
```

Comment:
afaict this deletes a document; but not a folder
for a folder you would need to traverse; just like the fullsync workflow but deleting instead of syncing (and no need to mark as deleted)

optionally we can move the deletion logic in the node resource (of course it would be legititmate to argue it's not in the direct pr scope)

Reply:
> I think the PR as is will not work (will not delete folders recursively)

Completely forgot the recursion, started this way but then totally forgot about it 🙄 Will traverse the nodes 👍🏼